### ISSUE TITLE
jekyll: linking to wrong version of python

### ISSUE THREAD

**Issue by mogorman** [NixOS/nixpkgs#12224]
When running jekyll in this fashion
```
#jekyll new test

#cd test

#jekyll serve

Configuration file: /home/mog/docs/code/test/_config.yml

            Source: /home/mog/docs/code/test

       Destination: /home/mog/docs/code/test_site

      Generating... 

  Liquid Exception: Failed to get header. in _posts/2016-01-07-welcome-to-jekyll.markdown
```

when searching the error online it seems to be that its an incompatible version of python https://teamtreehouse.com/community/error-running-jekyll-serve-liquid-exception-failed-to-get-header
it seems that they only support python2.7.  I looked at the nix and didnt see how it was pulling in python.

**Comment by jgillich** [comment]
http://jekyllrb.com/docs/templates/ "Code snippet highlightingPermalink"

I'd guess you have pygments installed somewhere, pygments.rb detects it and uses the wrong Python version. Or something like that?

I'm looking into updating to jekyll 3, where rouge is the default.

**Comment by jgillich** [comment]
What's really weird about this is that, by default, there should be no syntax highlighter enabled.

**Comment by mogorman** [comment]
i looked into it more and it looks like pygments.rb has a really screwy way to look for python pygments.  so even when i added it to the nix expression it didnt function

**Comment by mogorman** [comment]
installing rouge and editing the _config.yml does solve my immediate problem

**Comment by jgillich** [comment]
Should be fixed by #12226

**Comment by mogorman** [comment]
was fixed by #12227 but a better long term fix is still #12226
